zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/Schema.scala

```diff
@@ -184,19 +184,31 @@ trait Measurement {
 
 //TODO when UI get the schema, it should know which is dimension/measure, functions that can apply onto it, etc.
 // so that it won't ask for a inapplicable function such that get the max to a string field
-case class Schema(typeName: String,
-                  dimension: Seq[Field],
-                  measurement: Seq[Field],
-                  primaryKey: Seq[Field],
-                  timeField: TimeField
-                 ) {
+abstract class Schema(typeName: String,
+                      dimension: Seq[Field],
+                      measurement: Seq[Field],
+                      primaryKey: Seq[Field]
+                     ) {
```

Comment:
IMO,  if we need to throw some exception in the implementation, that means the interface design is not correct. The `timeField` by nature is the property of the `Temporal` schema. 

In most of the code after query parsing, the parameter should be of type `TemporalSchema` only if it relies on the `timeField`.